triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala

```diff
@@ -849,23 +860,72 @@ private[lf] class Runner(
       acs: Seq[CreatedEvent],
       offset: LedgerOffset,
       msgFlow: Graph[FlowShape[TriggerMsg, TriggerMsg], T] = Flow[TriggerMsg],
-      name: String = "",
   )(implicit
       materializer: Materializer,
       executionContext: ExecutionContext,
   ): (T, Future[SValue]) = {
     val source =
       msgSource(client, offset, trigger.heartbeat, parties, transactionFilter)
+
     Flow
       .fromGraph(msgFlow)
-      .viaMat(getTriggerEvaluator(name, acs))(Keep.both)
+      .viaMat(getTriggerEvaluator(acs))(Keep.both)
       .via(submitOrFail)
       .join(source)
       .run()
   }
 }
 
-object Runner extends StrictLogging {
+object Runner {
+
+  import Implicits._
+
+  private[trigger] val logger = ContextualizedLogger.get(getClass)
+
+  def apply(
+      compiledPackages: CompiledPackages,
+      trigger: Trigger,
+      triggerConfig: TriggerRunnerConfig,
+      client: LedgerClient,
+      timeProviderType: TimeProviderType,
+      applicationId: ApplicationId,
+      parties: TriggerParties,
+  )(implicit loggingContext: LoggingContextOf[Trigger]): Runner = {
+    val setupId: UUID = UUID.randomUUID()
```

Comment:
In a future PR, we'll pass this into the `Runner` class so that a child/parent relationship can be logged